web-api/src/mind.rs

```diff
@@ -166,6 +168,44 @@ where
         .collect()
 }
 
+// struct that represents config of hyperparameters for the persona based benchmark
+#[derive(Debug, Deserialize)]
+struct PersonaBasedConfig {
+    click_probability: f32,
+    n_documents: usize,
+    iterations: usize,
+    amount_of_doc_used_to_prepare: usize,
+    nranks: Vec<i32>,
+}
+
+impl PersonaBasedConfig {
+    fn new(
+        click_probability: f32,
+        n_documents: usize,
+        iterations: usize,
+        amount_of_doc_used_to_prepare: usize,
+        nranks: Vec<i32>,
+    ) -> Self {
+        Self {
+            click_probability,
+            n_documents,
+            iterations,
+            amount_of_doc_used_to_prepare,
+            nranks,
+        }
+    }
+
+    fn default() -> Self {
```

Comment:
🔧 for default impls the `Default` trait should be used to make it usable in other default contexts
```suggestion
    }
}

impl Default for PersonaBasedConfig {
    fn default() -> Self {
```